### ISSUE TITLE
P4: Lfm2MoeRouter and Qwen3MoeRouter softmax precise=True discrepancy lacks inline code comments

### ISSUE THREAD

**Issue by egesabanci** [egesabanci/reap-mlx#78]
## Summary

`Lfm2MoeRouter.__call__` (`router.py:199-233`) and `Qwen3MoeRouter.__call__` (`router.py:105-143`) differ in how they apply `mx.softmax`. The Qwen3 router uses `precise=True` while the LFM2 router does not:

**Qwen3:**
```python
gates = mx.softmax(logits, axis=-1, precise=True)
```

**LFM2:**
```python
logits = self.gate(hidden_states).astype(mx.float32)
gates = mx.softmax(logits, axis=-1)
```

This is **intentional** per architecture semantics (documented in `docs/observation-and-metrics.md`), but there is no inline code comment explaining *why* the difference matters. A future reader or contributor may "fix" the discrepancy, silently changing LFM2 routing behavior.

## Impact

- **Refactoring risk.** A well-intentioned contributor adding `precise=True` to the LFM2 router to "match" Qwen3 would change the routing behavior.
- **Onboarding friction.** A developer reading both routers sees an unexplained difference and must dig into architecture docs or git history to understand why.
- **Testing gap.** The router reference functions in `test_mlx_router.py` (`qwen_reference` and `lfm2_reference`) replicate the same discrepancy with no comment.

Severity is P4 (documentation/code clarity only, no functional bug).

## Root Cause

The Qwen3-MoE architecture specifies `precise=True` in its MLX forward pass; the LFM2 architecture does not. The difference was carried into the router implementations without an explanatory comment.

## Expected Behavior

Both router `__call__` methods should have a comment explaining the `precise` parameter choice, with a forward reference to the documentation that justifies the difference.

### Add comments to both routers

`src/reap/router.py`, in `Qwen3MoeRouter.__call__`:
```python
# Qwen3-MoE architecture uses precise softmax.
# See docs/observation-and-metrics.md § Qwen3 Router Semantics.
gates = mx.softmax(logits, axis=-1, precise=True)
```

`src/reap/router.py`, in `Lfm2MoeRouter.__call__`:
```python
# LFM2 architecture does NOT use precise softmax.
# This is intentional and matches the LFM2-MLX forward pass.
# See docs/observation-and-metrics.md § LFM2 Router Semantics.
gates = mx.softmax(logits, axis=-1)
```

### Same for the reference test functions

`tests/test_mlx_router.py`:
```python
def qwen_reference(mx, mlp, hidden_states, top_k, norm_topk_prob):
    # Qwen3 uses precise=True -- see router.py comments
    ...
    gates = mx.softmax(logits, axis=-1, precise=True)
    ...

def lfm2_reference(mx, moe, hidden_states, top_k, norm_topk_prob):
    # LFM2 does not use precise -- see router.py comments
    ...
    gates = mx.softmax(logits, axis=-1)
    ...
```

## Affected Code Paths

| File | Line | Role |
|---|---|---|
| `src/reap/router.py` | 126 | `Qwen3MoeRouter.__call__` -- `precise=True`, no comment |
| `src/reap/router.py` | 216 | `Lfm2MoeRouter.__call__` -- no `precise`, no comment |
| `tests/test_mlx_router.py` | (reference fns) | Both reference functions -- same discrepancy, no comment |
| `docs/observation-and-metrics.md` | Qwen3/LFM2 sections | Documents the difference |

## Notes

- This is purely a code clarity issue. The current behavior is correct.
- The test `test_router_normalization_epsilon_is_shared` already verifies `_NORM_EPSILON` is shared between both routers, confirming the epsilon is not the difference.
- The Qwen3 router also differs in flattening hidden states before the gate call (another intentional architecture difference that should ideally be commented).

**Comment by egesabanci** [comment]
Fixed in PR #82 (merged in commit 0f5c5cf). Closing as implemented.
